### ISSUE TITLE
The coverage plugin should maintain a pre-plogin hook to execute needed coverage generation

### ISSUE THREAD

**Issue by BioPhoton** [code-pushup/cli#462]
### User story

As a plugin user I want to be able to maintain my code coverage generation within my plugin config.
The coverage plugin should maintain a pre-plogin hook to execute needed coverage generation code e.g. `nx run cli:unit-test --coverage`

### Acceptance criteria

- [ ] The plugin options provide a per-execution hook or so to provide the scrip that generated the coverage files
- [ ] The script fails gracefully and the CLI continues
- [ ] The provided script is executed before collecting the report

**Comment by BioPhoton** [comment]
This is a example why we need a `table` section in the audit `details`.

<img width="766" alt="Screenshot 2024-02-06 at 00 23 20" src="https://github.com/code-pushup/cli/assets/10064416/0b638a0c-5058-4300-a87e-f0629fd90889">

**Comment by Tlacenka** [comment]
I am not sure if we need a table here. All missing coverage information is parsed into audit issues within the plugin and displayed in the report. And people usually look for overall coverage %, not per-file coverage. 🤔

**Comment by BioPhoton** [comment]
It is a visual overview. in parallel to the issues.

**Comment by Tlacenka** [comment]
This issue is a duplicate of AC in #430. Additionally, this part of the AC has been implemented recently.
Therefore, this issue can be closed.
